File: android/vbt-app/app/src/main/java/com/vbt/app/domain/model/ExerciseType.kt
package com.vbt.app.domain.model

enum class ExerciseType(
    val displayName: String,
    val category: String,
    val defaultMinLiftVelocity: Float,
    val defaultEndLiftVelocity: Float,
    val defaultMinRepDistance: Float,
    val mvt: Float // Minimum Velocity Threshold for 1RM estimation
) {
    BACK_SQUAT("Back Squat", "squat", 0.08f, 0.04f, 0.20f, 0.30f),
    FRONT_SQUAT("Front Squat", "squat", 0.08f, 0.04f, 0.20f, 0.30f),
    BENCH_PRESS("Bench Press", "bench", 0.12f, 0.05f, 0.15f, 0.17f),
    OVERHEAD_PRESS("Overhead Press", "ohp", 0.10f, 0.05f, 0.20f, 0.20f),
    DEADLIFT("Deadlift", "deadlift", 0.08f, 0.04f, 0.30f, 0.15f),
    ROMANIAN_DEADLIFT("Romanian Deadlift", "deadlift", 0.06f, 0.03f, 0.20f, 0.15f),
    CUSTOM("Custom", "custom", 0.10f, 0.05f, 0.10f, 0.25f)
}

File: ios/VBT/VBT/Domain/Models/ExerciseType.swift
import Foundation

/// Port 1:1 z Android `domain/model/ExerciseType.kt`.
enum ExerciseType: String, CaseIterable {
    case backSquat, frontSquat, benchPress, overheadPress, deadlift, romanianDeadlift, custom

    struct Defaults {
        let displayName: String
        let category: String
        let minLiftVelocity: Float
        let endLiftVelocity: Float
        let minRepDistance: Float
        /// Minimum Velocity Threshold — używane do estymacji 1RM.
        let mvt: Float
    }

    var defaults: Defaults {
        switch self {
        case .backSquat:
            return Defaults(displayName: "Back Squat", category: "squat", minLiftVelocity: 0.08, endLiftVelocity: 0.04, minRepDistance: 0.20, mvt: 0.30)
        case .frontSquat:
            return Defaults(displayName: "Front Squat", category: "squat", minLiftVelocity: 0.08, endLiftVelocity: 0.04, minRepDistance: 0.20, mvt: 0.30)
        case .benchPress:
            return Defaults(displayName: "Bench Press", category: "bench", minLiftVelocity: 0.12, endLiftVelocity: 0.05, minRepDistance: 0.15, mvt: 0.17)
        case .overheadPress:
            return Defaults(displayName: "Overhead Press", category: "ohp", minLiftVelocity: 0.10, endLiftVelocity: 0.05, minRepDistance: 0.20, mvt: 0.20)
        case .deadlift:
            return Defaults(displayName: "Deadlift", category: "deadlift", minLiftVelocity: 0.08, endLiftVelocity: 0.04, minRepDistance: 0.30, mvt: 0.15)
        case .romanianDeadlift:
            return Defaults(displayName: "Romanian Deadlift", category: "deadlift", minLiftVelocity: 0.06, endLiftVelocity: 0.03, minRepDistance: 0.20, mvt: 0.15)
        case .custom:
            return Defaults(displayName: "Custom", category: "custom", minLiftVelocity: 0.10, endLiftVelocity: 0.05, minRepDistance: 0.10, mvt: 0.25)
        }
    }
}
